packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -1383,6 +1383,16 @@ export default abstract class SqlIntegration
 
     return format(sqlQuery, this.getFormatDialect());
   }
+  getExistingMetricId(
+    existingMetrics: MetricInterface[],
+    type: MetricType,
+    sql: string
+  ): string {
+    const existingMetric = existingMetrics.find(
+      (metric) => metric.type === type && metric.sql === sql
```

Comment:
Do I really need the type here - should the sql cover that?

This does feel pretty fragile as any change to the sql would mean it's not a match. Should I also do the `metric.name`?

Ideally, there'd be a way to know if a metric was derived from a tracked event, and use that, but I don't think that'll be feasible.